packages/cli-core/src/printer/printer.ts

```diff
@@ -103,36 +107,54 @@ export class Printer {
   };
 
   /**
-   * Start animating ellipsis at the end of a log message.
+   * Writes escape sequence to stdout
    */
-  private startAnimatingEllipsis(message: string) {
+  private writeEscapeSequence(action: EscapeSequence) {
+    if (!this.isTTY()) {
+      return;
+    }
+
+    this.stdout.write(action);
+  }
+
+  /**
+   * Checks if the environment is TTY
+   */
+  private isTTY() {
+    return this.stdout.isTTY;
+  }
+
+  /**
+   * Start animating spinner at the end of a log message.
+   */
+  private startAnimatingSpinner(message: string) {
     if (!this.isTTY()) {
       this.log(message, LogLevel.INFO);
       return;
     }
 
     if (this.timerSet) {
```

Comment:
This isn't technically part of this PR, but there's a subtle inconsistency here where if stdout is a TTY, it will allow multiple timers running at once (because of the return above). Instead we should check for timerSet first, then isTTY so that the behavior is consistent in both environments.

Reply:
Just realized this isn't actually a problem because if it's a TTY, it will never get to the point of setting the timer in the first place. But it's fine, you don't need to change it back